apps/daimo-mobile/src/action/useSendAsync.ts

```diff
@@ -132,16 +153,27 @@ async function sendAsync(
   }
 }
 
-async function requestEnclaveSignature(enclaveKeyName: string, hexTx: string) {
+export async function requestEnclaveSignature(
+  enclaveKeyName: string,
+  hexTx: string,
+  usageMessage: string,
+  forceWeakerKeys: boolean
```

Comment:
so i think we now need `forceWeakerKeys` everywhere we use `enclaveKeyName`. i wonder if it's worth extracting an `EnclaveKeyInfo` that looks something like

```
interface EnclaveKeyInfo {
  keyName: string
  keyType: <"software" | "hardware">
}
```

Reply:
bit less code in a lot of places, eg here we'd pass `keyInfo: EnclaveKeyInfo`

since figuring out the `keyType` on existing accounts might be inconvenient in the migration, you could also say

```ts
interface EnclaveKeyInfo {
  keyName: string
  forceWeakerKeys: boolean
}
```

Reply:
done latter since easier + also technically we only need to know whether we need to force the enclave or not. If not we can save a call as it can autodetect key being software or hardware.